lib/lambda/processEmails.ts

```diff
@@ -130,25 +130,53 @@ export const handler: Handler<KafkaEvent> = async (event) => {
 
 export async function processRecord(kafkaRecord: KafkaRecord, config: ProcessEmailConfig) {
   const { key, value, timestamp } = kafkaRecord;
+
+  if (typeof key !== "string") {
+    console.log("key is not a string ", JSON.stringify(key, null, 2));
+    throw new Error("Key is not a string");
+  }
+
   const id: string = decodeBase64WithUtf8(key);
+  const safeID = id.replace(/^"|"$/g, "").toUpperCase();
+
+  if (!value) {
+    console.log("Tombstone detected. Doing nothing for this event");
+    return;
+  }
+  const valueParsed = JSON.parse(decodeBase64WithUtf8(value));
```

Comment:
I think `parsedValue` is a more "active" variable name than passive `valueParsed`

Reply:
I know you didn't write the original, but perhaps we can leave it better than we found it